### DISCUSSION TITLE
Two-Factor Authentication - How to configure

### DISCUSSION THREAD

**Discussion by pmatuszy** [healthchecks/healthchecks#749] [General]
Hello
a question: 
how to configure TFA? I found this
https://github.com/healthchecks/healthchecks#two-factor-authentication
 but this is not obvious.
RP_ID - is it environment variable? 
Regards
Paul

**Comment by cuu508** [comment]
It is a setting in `settings.py`. `settings.py` does read values from environment variables (see https://healthchecks.io/docs/self_hosted_configuration/), so, yes, you can pass it as an environment variable.

**Reply by pmatuszy** [reply]
ok - great. I thought it can be passed as env. I set it to my hostname.domainame, restarted gunicorn but I can't see any difference. Is there anything else I should do to get TFA working?
Paul

**Reply by cuu508** [reply]
Setting `RP_ID` enables WebAuthn support in your Healthchecks instance. The next step is to register your security keys. In Healthchecks web UI, go to Account Settings > Two-factor authentication. There should now be a "Add Security Key" button (which is absent if RP_ID is not set).

**Reply by pmatuszy** [reply]
wow - it works! thank you. That was straightforward. 
Thank you!
